dynamic_library/tests/dynamic_library_tests.cc

```diff
@@ -0,0 +1,78 @@
+#include <catch.hpp>
+#include <leatherman/dynamic_library/dynamic_library.hpp>
+
+namespace leatherman { namespace dynamic_library {
+
+#ifdef _WIN32
+    std::string const lib_path = "libtest.so";
+    std::string const lib_path2 = "libtest1.so";
+#else
+    std::string const lib_path = "../lib/libtest.so";
+    std::string const lib_path2 = "../lib/libtest1.so";
+#endif
+
+    TEST_CASE("dynamic_library::load and dynamic_library::close", "[dyn-lib]"){
+        SECTION("should not be loaded by default") {
+            dynamic_library lib;
+            REQUIRE_FALSE(lib.loaded());
+        }
+
+        SECTION("should load library from path, then close it"){
+            dynamic_library lib;
+            REQUIRE(lib.load(lib_path));
+            REQUIRE(lib.loaded());
+            REQUIRE(lib.first_load());
+            lib.close();
+            REQUIRE_FALSE(lib.loaded());
+        }
+    }
+
+    TEST_CASE("dynamic_library::find_symbol", "[dyn-lib]"){
+        dynamic_library lib;
+        REQUIRE(lib.load(lib_path));
+
+        SECTION("should fail to find nonexistent symbol"){
+            REQUIRE_FALSE(lib.find_symbol("not_here"));
+        }
+
+        SECTION("should find library function"){
+            REQUIRE(lib.find_symbol("hello"));
+        }
+
```

Comment:
We should add a test exercising find_symbol with `throw_if_missing=true`; and one that exercises alias with `alias=goodbye` on libtest1.